mmv1/third_party/terraform/services/storage/resource_storage_bucket.go.tmpl

```diff
@@ -1058,6 +1058,19 @@ func resourceStorageBucketDelete(d *schema.ResourceData, meta interface{}) error
 		// Wait for everything to finish.
 		wp.StopWait()
 	}
+	// get all folders and subfolders using List API folder
+	folder, _ := config.NewStorageClient(userAgent).Folders.List(bucket).Do()
+	var folderItems []*storage.Folder
+	if folder.Items != nil && len(folder.Items) > 0 {
+		folderItems = folder.Items
+	}
+	// delete folders from bottom to top to avoid no empty folder error
+	for j := len(folderItems) - 1; j >= 0; j-- {
```

Comment:
Why would you not use the similar mechanism like above of workerpool?
Retention policy check, listing error, etc?

Reply:
this is not needed, I removed this logic, there is some problem with the manually created folders on GCP console and creating folders through API,

when manually created there are some invisible files and so the bucket is not empty and the bucket not deleting.

I added a testcase with nested folders and uploading an object, the testcase is working fine and I verified through API's also

Reply:
Log a separate item. The UI creates a placeholder object i.e. an object and an object ending with a '/'. That should be handled outside the scope of the current feature is it is an existing integration.

Do you have tests which have a HNS bucket with only objects in a hierarchy and they get deleted? You can create objects in a hierarchy by creating an object ending in / and then adding objects under it for a non-folders use case and ensure that they do get all cleaned up.